webapps/frontend/app/viewmodels/wizards/createfromtemplatewizard/gather.js

```diff
@@ -155,6 +176,16 @@ define([
                 self.data.vm().load();
                 self.data.selectedPMachines([]);
             }
+            generic.xhrAbort(self.loadVMachinesHandle);
+            self.loadVMachinesHandle = api.get('vmachines/')
```

Comment:
You're not waiting for the vms to be loaded, so the validation you do when clicking next might be incorrect